nil/core-concepts/transaction-lifecycle.mdx

```diff
@@ -0,0 +1,51 @@
+# Transaction lifecycle
+
+## Definition
+
+In =nil;, a transaction passes the following stages before reaching finality.
+
+1. A transaction is submitted by an account
+2. (Optional) Validators track the transaction to determine the shard where the transaction is supposed to be included in a block
+3. The transaction goes to the mempool of an execution shard
+4. The transaction passes the 'searcher-builder-relayer' stages and the block with the transaction is sent to the proposer
+5. The block passes local consensus while the shard generates a ZKP
+6. The ZKP and the block are sent to the consensus shard that produces a 'master' ZKP
+7. The consensus sends shard a DA transaction to Ethereum and the transaction achieves soft finality
+8. Absolute finality is reached once all ZKPs are verified
+
+Additional remarks about each of the above stages are given below.
+
+## Stages
+
+### Transaction submission and processing
+
+Whenever a user submits a new transaction, it is added to the mempool of an execution shard. 
+
+If the transaction is an outgoing message (e.g., it relates to an account located on another shard), validators must track it across shards and ensure its inclusion in the new block of its destination shard.
```

Comment:
Validators of the outgoing shard do not track the message's path

Reply:
I see, amended